ui-v2/app/utils/dom/click-first-anchor.js

```diff
@@ -1,9 +1,15 @@
-const clickEvent = function() {
-  return new MouseEvent('click', {
-    bubbles: true,
-    cancelable: true,
-    view: window,
-  });
+const clickEvent = function($el) {
+  ['mousedown', 'mouseup', 'click']
+    .map(function(type) {
+      return new MouseEvent(type, {
+        bubbles: true,
+        cancelable: true,
+        view: window,
+      });
+    })
+    .forEach(function(event) {
+      $el.dispatchEvent(event);
```

Comment:
Given that this is bound to an `onclick` handler in the tabular collection, I can see this behavior being surprising. That a click event in one place would immediately dispatch a mousedown, mouseup, and click in another place.

Reply:
Not massively sure what you mean here, can you explain a little further? I kept the previous functionality but added the `mousedown`/`mouseup` event to make it more like a true `click` (and therefore the new functionality works here also)

Reply:
I'm thinking about this code here: https://github.com/hashicorp/consul/blob/master/ui-v2/app/components/tabular-collection.js#L284-L286

If I were to read this code, or change this code, I don't think I'd immediately know that in addition to the click event, the mousedown and mouseup events would be sent.

I don't mean to raise this as a warning flag or anything, just an idle thought about the abstraction.

Reply:
Hmmm I see what you mean, I think I'm ok with this as a `click` event 'means' `mousedown -> mouseup -> click` anyway, we just don't think of it like that. Therefore theoretically something called `clickFirstAnchor` to emulate clicking on the first anchor should probably call all three events.